src/components/atom/button/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { ReactComponent as AddIcon } from "../../../assets/img/addIcon.svg";
+import "./styles.scss";
+
+type ButtonProps = {
+  text: string;
+  hasAddIcon?: boolean;
+  onClick: () => void;
+  type: HTMLButtonElement["type"];
+  disabled?: boolean;
+  styles: string;
+};
+
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
+  return (
+    <button className={styles} onClick={onClick} disabled={disabled} type={type} {...rest}>
+      {hasAddIcon && <AddIcon />}
```

Comment:
Also, try to avoid {hasAddIcon && <AddIcon />}. Rather use {hasAddIcon ? <AddIcon /> : null} This is one of the most common mistakes. React does not render "false" which will be the first case. This can lead to a small performance overhead if the component tree is complex and deeply nested.